scripts/cli-smoke-native.py

```
@@ -0,0 +1,114 @@
+
+def _resolve(name: str) -> str:
+    """Resolve an executable on PATH (finds console scripts, .exe on Windows)."""
+    path = shutil.which(name)
+    if path is None:
+        raise RuntimeError(f"executable {name!r} not found on PATH")
+    return path
+
+
+def _run(cmd: list[str], cwd: str | None = None) -> str:
+    """Run a command, fail loud on non-zero, return combined stdout+stderr."""
+    proc = subprocess.run(cmd, check=True, capture_output=True, text=True, cwd=cwd)
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)